src/components/organisms/NewResourceWizard/NewResourceWizard.tsx

```diff
@@ -103,26 +115,74 @@ const NewResourceWizard = () => {
       return;
     }
 
+    createResourceProcessing();
+
+    closeWizard();
+  };
+
+  const getFullFileName = (filename: string) => {
+    if (filename.endsWith('.yaml') || filename.endsWith('.yml')) {
+      return filename;
+    }
+
+    return `${filename}.yaml`;
+  };
+
+  const createResourceProcessing = () => {
+    const formValues = form.getFieldsValue();
+
     const selectedResource =
-      data.selectedResourceId && data.selectedResourceId !== SELECT_OPTION_NONE
-        ? resourceMap[data.selectedResourceId]
+      formValues.selectedResourceId && formValues.selectedResourceId !== SELECT_OPTION_NONE
+        ? resourceMap[formValues.selectedResourceId]
         : undefined;
     const jsonTemplate = selectedResource?.content;
 
-    createUnsavedResource(
+    const newResource = createUnsavedResource(
       {
-        name: data.name,
-        kind: data.kind,
-        namespace: data.namespace === NO_NAMESPACE ? undefined : data.namespace,
-        apiVersion: data.apiVersion,
+        name: formValues.name,
+        kind: formValues.kind,
+        namespace: formValues.namespace === NO_NAMESPACE ? undefined : formValues.namespace,
+        apiVersion: formValues.apiVersion,
       },
       dispatch,
       jsonTemplate
     );
 
+    const fullFileName = getFullFileName(formValues.name);
+
+    if (shouldSaveToFolder) {
+      setTimeout(() => {
```

Comment:
could you simply add an optional savePath property to the createUnsavedResource method above - which would save the resource after creating it? (if the path was provided)

Reply:
Sure